infer/SigSuggestion.cc

```diff
@@ -426,14 +426,18 @@ bool SigSuggestion::maybeSuggestSig(core::Context ctx, core::ErrorBuilder &e, un
 
     ENFORCE(!methodSymbol.data(ctx)->arguments().empty(), "There should always be at least one arg (the block arg).");
     bool onlyArgumentIsBlkArg = methodSymbol.data(ctx)->arguments().size() == 1 &&
-                                methodSymbol.data(ctx)->arguments()[0].name == core::Names::blkArg();
+                                methodSymbol.data(ctx)->arguments()[0].isSyntheticBlockArgument();
 
     if (!onlyArgumentIsBlkArg) {
         fmt::format_to(ss, "params(");
 
         bool first = true;
         for (auto &argSym : methodSymbol.data(ctx)->arguments()) {
-            if (argSym.name == core::Names::blkArg()) {
+            // WARNING: This is doing raw string equality--don't cargo cult this!
+            // You almost certainly want to compare NameRef's for equality instead.
+            // We need to compare strings here because we're running with a frozen global state
+            // (and thus can't take the string that we get from `argumentName` and enter it as a name).
+            if (argSym.argumentName(ctx) == core::Names::blkArg().data(ctx)->shortName(ctx)) {
```

Comment:
@DarkDimius why shortName over show?

Reply:
`shortName` returns stringView and thus doesn't allocate.

can we do the NameRef comparison now?

Reply:
🤯

Reply:
@DarkDimius no because `.name` is still always `<blk>`. I think it's this way for override checking?

Reply:
https://github.com/sorbet/sorbet/blob/022a5a5ea471511c3252f8b8400613ddd8bef257/namer/namer.cc#L87-L88

Reply:
👍